flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -518,8 +519,14 @@ class OmpAttributeVisitor : DirectiveAttributeVisitor<llvm::omp::Directive> {
   }
 
   bool Pre(const parser::OmpClause::Reduction &x) {
-    const parser::OmpReductionIdentifier &opr{
-        std::get<parser::OmpReductionIdentifier>(x.v.t)};
+    const auto &objList{std::get<parser::OmpObjectList>(x.v.t)};
+    ResolveOmpObjectList(objList, Symbol::Flag::OmpReduction);
+
+    auto &modifiers{OmpGetModifiers(x.v)};
```

Comment:
It took me a bit to process that reduction identifiers are modifiers also. Was that established in OpenMP 5.2?

Reply:
In explicit terms, yes.  Before 5.2 it was a de facto modifier by the virtue of being on the left side of a `:`, e.g. `reduction(+: x, y, z)`.  In 5.2 all such things are modifiers (there are also a couple of post-modifiers, i.e. following a `:`).